lib/usd/translators/meshWriter.cpp

```diff
@@ -413,34 +371,116 @@ PxrUsdTranslators_MeshWriter::writeMeshAttrs(
         // not adding the clamp attribute as custom data. We also don't need to
         // reconstruct a color set from them on import since they originated
         // from the bound shader(s), so the authored flag is set to false.
-        _addDisplayPrimvars(primSchema,
-                            usdTime,
-                            MFnMesh::kRGBA,
-                            shadersRGBData,
-                            shadersAlphaData,
-                            shadersInterpolation,
-                            shadersAssignmentIndices,
-                            false,
-                            false);
+        UsdMayaMeshUtil::addDisplayPrimvars(primSchema,
+                                            usdTime,
+                                            MFnMesh::kRGBA,
+                                            shadersRGBData,
+                                            shadersAlphaData,
+                                            shadersInterpolation,
+                                            shadersAssignmentIndices,
+                                            false,
+                                            false,
+                                            *_GetSparseValueWriter());
     }
 
     return true;
 }
 
-/* virtual */
 bool
 PxrUsdTranslators_MeshWriter::ExportsGprims() const
 {
     return true;
 }
 
 bool
-PxrUsdTranslators_MeshWriter::_IsMeshAnimated() const
+PxrUsdTranslators_MeshWriter::isMeshAnimated() const
 {
     // Note that _HasAnimCurves() as computed by UsdMayaTransformWriter is
     // whether the finalMesh is animated.
-    return _skelInputMesh.isNull() ? _HasAnimCurves() : false;
+    return m_skelInputMesh.isNull() ? _HasAnimCurves() : false;
 }
 
+void
+PxrUsdTranslators_MeshWriter::cleanupPrimvars()
+{
+    if (!isMeshAnimated()) {
+        // Based on how SetPrimvar() works, the cleanup phase doesn't apply to
```

Comment:
I think this now actually refers to the `setPrimvar()` function that's in an anonymous namespace in `meshWriteUtils.cpp`? That's an unfortunate hidden coupling.

Reply:
See f7f193bcf49fd73724dc65643429fa556f7e825b